src/cc/bcc_proc.c

```diff
@@ -392,3 +392,74 @@ char *bcc_procutils_which_so(const char *libname, int pid) {
 void bcc_procutils_free(const char *ptr) {
   free((void *)ptr);
 }
+
+/* Detects the following languages + C. */
+const char *languages[] = {"java", "python", "ruby", "php", "node"};
+const int nb_languages = 5;
+
+char *bcc_procutils_language(int pid) {
+  char procfilename[22], line[4096], pathname[32], *str;
+  FILE *procfile;
+  int i, ret;
+
+  /* Look for clues in the process' command line. */
```

Comment:
Not through the command line check but both `/proc/$pid/exe` and `/proc/$pid/maps` will point to `ruby` when using `irb`. `rake` has the same behavior.

When using `jruby`, however, `detect_language` will fail to return `Java` (only because of the `/proc/$pid/cmdline` check). I will remove the `/proc/$pid/cmdline` and `/bin/$language` checks for now; we can always add them back latter if some language requires it.